### PR TITLE
docs: improve API docs of all crawlers

### PR DESCRIPTION
**Pull request by vdusek**
### Description

- Improve API docs of all crawlers.

### Issues

- Relates #304 

### Testing

- The doc website was rendered locally.

### Checklist

- [x] CI passed

### CODE REVIEW

File: src/crawlee/basic_crawler/_basic_crawler.py [hunk 1]
```diff
@@ -88,14 +88,27 @@ class BasicCrawlerOptions(TypedDict, Generic[TCrawlingContext]):
...
 class BasicCrawler(Generic[TCrawlingContext]):
-    """Provides a simple framework for parallel crawling of web pages.
-
-    The URLs to crawl are fed either from a static list of URLs or from a dynamic queue of URLs enabling recursive
-    crawling of websites.
-
-    `BasicCrawler` is a low-level tool that requires the user to implement the page download and data extraction
-    functionality themselves. If we want a crawler that already facilitates this functionality, we should consider using
-    one of its subclasses.
+    """A basic web crawler providing a framework for crawling websites.
```
Comment (janbuchar) [inline src/crawlee/basic_crawler/_basic_crawler.py:91]: Any chance we could make the class names links?

File: src/crawlee/http_clients/curl_impersonate.py [hunk 2]
```diff
@@ -77,6 +77,16 @@ class CurlImpersonateHttpClient(BaseHttpClient):
...
     See the `BaseHttpClient` class for more common information about HTTP clients.
+
+    ### Usage
+
+    ```python
+    from crawlee.http_clients.curl_impersonate import CurlImpersonateHttpClient
+    from crawlee.http_crawler import HttpCrawler  # or any other HTTP client-based crawler
+
+    http_client = CurlImpersonateHttpClient()
+    crawler = HttpCrawler(http_client=http_client)
```
Comment (janbuchar) [inline src/crawlee/http_clients/curl_impersonate.py:88]: Thanks for an example that's actually practical :slightly_smiling_face:

File: src/crawlee/http_crawler/_http_crawler.py [hunk 3]
```diff
@@ -15,7 +15,40 @@
...
-    """A crawler that fetches the request URL using `httpx`."""
+    """A web crawler for performing HTTP requests.
+
+    The `HttpCrawler` builds on top of the `BasicCrawler`, which means it inherits all of its features. On top
+    of that it implements the HTTP communication using the HTTP clients. The class allows integration with
+    any HTTP client that implements the `BaseHttpClient` interface. The HTTP client is provided to the crawler
+    as an input parameter to the constructor.
+
+    The HTTP client-based crawlers are ideal for websites that do not require JavaScript execution. However,
+    if you need to execute client-side JavaScript, consider using browser-based crawler like the `PlaywrightCrawler`.
```
Comment (janbuchar) [inline src/crawlee/http_crawler/_http_crawler.py:26]:
```suggestion
    if you need to execute client-side JavaScript, consider using a browser-based crawler like the `PlaywrightCrawler`.
```

**Review by janbuchar — APPROVED** [review]
I skimmed this and found some typos. I think it's pretty much good to go though, thanks for taking care of this.

File: src/crawlee/basic_crawler/_basic_crawler.py (hunk 1 above)
Comment (vdusek) [reply src/crawlee/basic_crawler/_basic_crawler.py:91]: I have no idea, opened a new issue for that -> https://github.com/apify/crawlee-python/issues/625.

**Pull request merged**